tests/unit/package_managers/bundler/test_main.py

```diff
@@ -1,10 +1,77 @@
+from unittest import mock
+
 import pytest
 from git.repo import Repo
 
 from cachi2.core.errors import PackageRejected
-from cachi2.core.package_managers.bundler.main import _get_main_package_name_and_version
-from cachi2.core.package_managers.bundler.parser import GemDependency, ParseResult, PathDependency
+from cachi2.core.package_managers.bundler.main import (
+    _get_main_package_name_and_version,
+    _resolve_bundler_package,
+)
+from cachi2.core.package_managers.bundler.parser import (
+    GemDependency,
+    GitDependency,
+    ParseResult,
+    PathDependency,
+)
 from cachi2.core.rooted_path import RootedPath
+from tests.common_utils import GIT_REF
+
+
```

Comment:
I'm missing a unit test verifying the PURL format, we have one like that (in context of creating an SBOM component) for `gomod` and `rpm` (likely other backends as well).